### PR TITLE
FPSCounterBundle

### PR DESCRIPTION
**Pull request by jojolepro**
Added FPSCounterBundle.
Ran rust updates + rustfmt update and formatted the code.

### CODE REVIEW

File: examples/04_pong/systems/winner.rs [hunk 1]
```diff
@@ -12,14 +12,12 @@ use audio::Sounds;
...
-    type SystemData = (
-        WriteStorage<'s, Ball>,
-        WriteStorage<'s, LocalTransform>,
-        FetchMut<'s, ScoreBoard>,
-        Fetch<'s, AssetStorage<Source>>,
-        Fetch<'s, Sounds>,
-        Fetch<'s, Option<Output>>,
-    );
+    type SystemData = (WriteStorage<'s, Ball>,
+     WriteStorage<'s, LocalTransform>,
```
Comment (torkleyy) [inline examples/04_pong/systems/winner.rs]: I feel like we change this formatting in every PR, and every contributor says they're using the latest version..

Comment (jojolepro) [reply examples/04_pong/systems/winner.rs]: rustup update
cargo +nightly install rustfmt --force
cargo +nightly fmt

Comment (torkleyy) [reply examples/04_pong/systems/winner.rs]: should be `cargo +nightly install --force rustfmt-nightly`

Comment (Rhuagh) [reply examples/04_pong/systems/winner.rs]: and `cargo +nightly fmt --all`

Comment (jojolepro) [reply examples/04_pong/systems/winner.rs]: Fixed

File: amethyst_utils/src/fps_counter.rs [hunk 2]
```diff
@@ -63,3 +63,29 @@ impl<'a> System<'a> for FPSCounterSystem {
...
+}
+impl FPSCounterBundle {
+    ///Creates a new FPSCounterBundle with the specified sample size.
+    pub fn new(samplesize: usize) -> Self {
+        Self {
+            samplesize: samplesize,
+        }
+    }
+    ///Same as FPSCounterBundle::new(20).
+    pub fn default() -> Self {
```
Comment (Xaeroxe) [inline amethyst_utils/src/fps_counter.rs]: We should instead impl Default trait for this structure

Comment (jojolepro) [reply amethyst_utils/src/fps_counter.rs]: Fixed ;)

File: amethyst_utils/src/fps_counter.rs [hunk 3]
```diff
@@ -63,3 +63,31 @@ impl<'a> System<'a> for FPSCounterSystem {
         //println!("Cur FPS: {}, Sampled: {}",counter.frame_fps(),counter.sampled_fps());
     }
 }
+///Automatically adds a FPSCounterSystem and a FPSCounter resource with the specified sample size.
+pub struct FPSCounterBundle {
+    samplesize: usize,
+}
```
Comment (Rhuagh) [inline amethyst_utils/src/fps_counter.rs]: Newline

File: amethyst_utils/src/fps_counter.rs [hunk 4]
```diff
@@ -63,3 +63,31 @@ impl<'a> System<'a> for FPSCounterSystem {
...
+impl FPSCounterBundle {
+    ///Creates a new FPSCounterBundle with the specified sample size.
+    pub fn new(samplesize: usize) -> Self {
+        Self {
+            samplesize: samplesize,
+        }
+    }
+}
```
Comment (Rhuagh) [inline amethyst_utils/src/fps_counter.rs]: See above

File: amethyst_utils/src/fps_counter.rs [hunk 5]
```diff
@@ -63,3 +63,31 @@ impl<'a> System<'a> for FPSCounterSystem {
...
+impl Default for FPSCounterBundle {
+    ///Same as FPSCounterBundle::new(20).
+    fn default() -> Self {
+        Self::new(20)
+    }
+}
```
Comment (Rhuagh) [inline amethyst_utils/src/fps_counter.rs]: .

File: amethyst_utils/src/fps_counter.rs [hunk 6]
```diff
@@ -63,3 +63,31 @@ impl<'a> System<'a> for FPSCounterSystem {
         //println!("Cur FPS: {}, Sampled: {}",counter.frame_fps(),counter.sampled_fps());
     }
 }
+///Automatically adds a FPSCounterSystem and a FPSCounter resource with the specified sample size.
```
Comment (Rhuagh) [inline amethyst_utils/src/fps_counter.rs]: Newline

File: amethyst_utils/src/fps_counter.rs [hunk 7]
```diff
@@ -1,9 +1,9 @@
 //! Util Resources
 
+use amethyst_core::{ECSBundle, Result};
 use amethyst_core::timing::{duration_to_nanos, Time};
-use specs::{Fetch, FetchMut, System};
-
 use circular_buffer::CircularBuffer;
+use specs::{DispatcherBuilder, Fetch, FetchMut, System, World};
```
Comment (torkleyy) [inline amethyst_utils/src/fps_counter.rs]: Why did you move this import?

Comment (jojolepro) [reply amethyst_utils/src/fps_counter.rs]: rustfmt?

Comment (torkleyy) [reply amethyst_utils/src/fps_counter.rs]: But it was in the other import group, rustfmt doesn't move betwen import groupts.

**Comment by torkleyy** [comment]
Thanks @jojolepro :+1: 

bors r+

File: amethyst_utils/src/fps_counter.rs (hunk 7 above)
Comment (Rhuagh) [reply amethyst_utils/src/fps_counter.rs]: He ran with normal rustfmt first, not rustfmt-nightly, that might have messed it up.

**Comment by jojolepro** [comment]
:tada:
